lib/manageiq/providers/ibm_cloud/cloud_tools/common/proxy.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+module ManageIQ
+  module Providers
+    module IbmCloud
+      module CloudTools
+        module Common
+          # Set proxy configurations dynamically.
+          class Proxy
+            ACCESSOR_KEYS = %i[address port username password headers].freeze
+            attr_accessor(*ACCESSOR_KEYS)
+
+            # Add a server configuration.
+            # @return [Proxy]
+            def server(address:, port:)
+              @address = address
+              @port = port
+              self
+            end
+
+            # Add a proxy authentication to the server configuration.
+            # @return [Proxy]
+            def authentication(username:, password:)
+              @username = username
+              @password = password
+              self
+            end
+
+            # Convert non-nil keys into a hash.
+            # @return [Hash]
+            def to_hash
+              self.class::ACCESSOR_KEYS.each_with_object({}) { |key, obj| obj[key] = instance_variable_get("@#{key}") }.compact # rubocop:disable Rails/IndexWith
```

Comment:
Whoa, TIL about `index_with`...that's *awesome*, and we should totally use that.

```suggestion
              ACCESSOR_KEYS.index_with { |k| instance_variable_get("@#{key}") }.compact
```

Reply:
It is a cool method https://sebastian-palma.github.io/2020-06-08/rails-index-by-index-with. However, 'index_with' is added by rails. I don't want to use a non-standard method.

Reply:
Given this plugin is a rails engine having rails is a pretty safe bet but I don't think we should hold this up for this

Reply:
Rails (active_support) is guaranteed

Reply:
I use CloudTool without the rest of rails to verify what I expect to work does work, this would hinder that ability.

Reply:
We do want to be able to run e.g. the collectors without rails :+1: 
That said requiring the active_support gem is pretty common and doesn't incur the same overhead that all of rails does.